src/parachain/amm.ts

```diff
@@ -0,0 +1,925 @@
+
+        const withdrawalExtrinsic = this.api.tx.dexGeneral.removeLiquidity(
+            newCurrencyId(this.api, pool.token0),
+            newCurrencyId(this.api, pool.token1),
+            amount.toString(true),
+            minAmounts[0],
+            minAmounts[1],
+            recipientAccount,
+            deadline
+        );
+
+        return [withdrawalExtrinsic, this.api.events.dexGeneral.LiquidityRemoved];
+    }
+
+    private async _getLiquidityWithdrawalStablePoolParams(
+        amount: MonetaryAmount<StableLpToken>,
+        pool: StableLiquidityPool,
+        maxSlippageComplement: number,
+        recipient: AddressOrPair,
+        deadline: number
+    ): Promise<[SubmittableExtrinsic<ApiTypes>, AugmentedEvent<ApiTypes>]> {
+        const outputAmounts = pool.getLiquidityWithdrawalPooledCurrencyAmounts(amount);
+        const minAmounts = outputAmounts.map((amount) => amount.mul(maxSlippageComplement));
+        const poolId = pool.poolId;
+        const lpTokenAmount = amount.toString(true);
+        const recipientAccount = addressOrPairAsAccountId(this.api, recipient);
+
+        if (isStableMetaPool(pool)) {
+            const basePoolId = pool.basePool.poolId;
+            const minAmountsMeta = pool.actuallyPooledCurrencies
+                .map(
+                    (actuallyPooledCurrency) =>
+                        minAmounts.find((amount) =>
+                            isCurrencyEqual(amount.currency, actuallyPooledCurrency.currency)
+                        ) || new MonetaryAmount(actuallyPooledCurrency.currency, 0)
+                )
+                .map(monetaryAmountToRawString);
+            // const minAmountsMeta = minAmounts
+            //     .filter((amount) => pool.involvesToken(amount.currency))
+            //     .map(monetaryAmountToRawString);
+            const minAmountsBase = minAmounts
+                .filter((amount) => pool.basePool.involvesToken(amount.currency))
+                .map(monetaryAmountToRawString);
+
+            const withdrawLiquidityExtrinsic = this.api.tx.dexStable.removePoolAndBasePoolLiquidity(
+                poolId,
+                basePoolId,
+                lpTokenAmount,
+                minAmountsMeta,
+                minAmountsBase,
+                recipientAccount,
+                deadline
+            );
+
+            return [withdrawLiquidityExtrinsic, this.api.events.dexStable.RemoveLiquidity];
+        }
+
+        const minAmountsRaw = minAmounts.map(monetaryAmountToRawString);
+        const withdrawLiquidityExtrinsic = this.api.tx.dexStable.removeLiquidity(
+            poolId,
+            lpTokenAmount,
+            minAmountsRaw,
+            recipientAccount,
+            deadline
+        );
+
+        return [withdrawLiquidityExtrinsic, this.api.events.dexStable.RemoveLiquidity];
+    }
+
+    async removeLiquidity(
+        amount: MonetaryAmount<LpCurrency>,
+        pool: LiquidityPool,
+        maxSlippage: number, // Percentage.
```

Comment:
Won't block on this, but just wondering if we could type "Percentage" to make it explicit what value should be passed?